### PR TITLE
Reduce unnecessary queries

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
@totten  what is your perception of the safety (risks) of these?

### CODE REVIEW

**Comment by colemanw** [comment]
If CRM_Utils_Cache::get hits the database then it's not much of an improvement.

**Comment by eileenmcnaughton** [comment]
The Utils_Cache function seems to use an array cache by default & memCache on config. On my local (no config) that bit of code is dropping around ? 70 queries? from a search page load?

**Comment by colemanw** [comment]
I'll let @totten comment about which cache function should be used since there are many.

**Comment by eileenmcnaughton** [comment]
Clearly the custom data patch needs more work - also @josephlacey apparently DOES extend email with custom data in an extension  - so assumptions = flawed.

File: CRM/Core/BAO/WordReplacement.php [hunk 1]
```diff
@@ -303,13 +303,21 @@ public static function getLocaleCustomStrings($locale, $domainId = NULL) {
...
+      ));
+      if (empty($domainStrings)) {
+        $cached = array();
+      }
+      else {
+        $cached = unserialize($domainStrings);
+      }
+      CRM_Utils_Cache::singleton()->set($cacheString, $cached);
     }
+    return ($cached) ? array() : $cached;
```
Comment (totten) [inline CRM/Core/BAO/WordReplacement.php]: The funny thing is that this data is already supposed to be cached (`CRM_Core_I18n::crm_translate_raw` calls `getLocaleCustomStrings` and stores it in `Civi::$statics`). But it looks like that caching wasn't working for blank locales.

This patch fixes that other cache and is simpler: https://github.com/civicrm/civicrm-core/pull/7214

**Comment by eileenmcnaughton** [comment]
Just noting that this relates https://issues.civicrm.org/jira/browse/CRM-15656

**Pull request closed without merging**